semantic_search/ncbi.py

```diff
@@ -99,8 +100,7 @@ def _medline_to_docs(records: List[Dict[str, str]]) -> List[Dict[str, str]]:
     docs = []
     for record in records:
         if "PMID" not in record:
-            logging.warn(f"No PMID for {json.dumps(record)}") 
-            continue 
+            raise HTTPException(status_code=422, detail=f"No PMID for {json.dumps(record)}") 
```

Comment:
Think we could parse out just the pmid from `json.dumps(record)` or `record`? Then we can write a more informative error here, e.g. (`f"pmid: {pmid} provided in the request is invalid"`).

Reply:
In that if block, the `record` variable has this inside it only (doesn't hold the `pmid` anywhere):

![image](https://user-images.githubusercontent.com/57552053/115073982-11d84a00-9ec7-11eb-8224-988d26467f80.png)

So I don't think we can just parse it out here, but maybe we could return it back to the `uids_to_docs` function with a flag and possibly get the `pmid` from just there, but I'm not sure. Or to keep it simple we can just regex the number out but that could cause issues maybe.

Reply:
I see. In that case I think it should just be:

```python
raise HTTPException(status_code=422, detail=json.dumps(record))
```

Because the `"No PMID for `" is redundant.

Reply:
Yeah for sure, I'll push it right now.

Reply:
Just noticing now that it might even be best to set `detail=json.dumps(record)["record"]["id"][-1]` to get right to the string `"Error occurred: The following PMID...."`

Reply:
@Anwesh1

Reply:
Oh sorry, I didn't see that comment before. This won't work because `record` doesn't have any other distinguishing tags inside it. Running this causes a `TypeError`: `raise HTTPException(status_code=422, detail=json.dumps(record)["record"]["id"])
TypeError: string indices must be integers`

Reply:
I may have gotten the exact syntax wrong, but presumably, there is a way to get to the actual string: `"XXXXXXXX Error occurred: The following PMID...."`. I am suggesting we extract that and pass it as the argument to `detail`.

Reply:
`record` only has one tag inside it ("id"). I have updated the code to access that:
```python
if "PMID" not in record:
            raise HTTPException(status_code=422, detail=json.dumps((record["id:"])))
```